third_party/amd/lib/TritonAMDGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandHelper.cpp

```diff
@@ -8,8 +8,8 @@ namespace mlir::triton::AMD {
 Value getWarpIdInBlock(ConversionPatternRewriter &rewriter, Location loc,
                        Value warpId, const ArrayRef<unsigned int> &wpt,
                        int elemPerInstrNonK, int tensorSizeNonK, int nonKIdx) {
-  if (nonKIdx == 1)
-    warpId = udiv(warpId, i32_val(wpt[0]));
+  if (nonKIdx == 0)
```

Comment:
Is this assuming the `order` of the tensor is always [1, 0]?

Reply:
> Is this assuming the `order` of the tensor is always [1, 0]?

Good point, I forgot to take into account transpose=True case. Fixing it.